src/state/opamSysInteract.ml

```diff
@@ -1102,14 +1114,59 @@ let install_packages_commands_t ?(env=OpamVariable.Map.empty) config sys_package
     [`AsUser (Commands.msys2 config),
      "-Su"::"--noconfirm"::packages], None
   | Netbsd -> [`AsAdmin "pkgin", yes ["-y"] ("install" :: packages)], None
+  | Nix ->
+    (match st with
+     | None ->
+       log "Nix depext must be passed switch";
+       [], None
+     | Some (st : _ OpamStateTypes.switch_state) ->
+       let dir = OpamPath.Switch.meta st.switch_global.root st.switch in
+       let drvFile =
+         OpamFilename.create dir
+           (OpamFilename.basename (OpamFilename.raw "env.nix"))
```

Comment:
This file seems to be there only as a sort of temporary file for `nix-build`. I also find the naming convention a bit confusing when taken together with `nix.env`.
`env.nix` and `nix.env` are way too close in terms of name and it's easy to confuse between them.

Wouldn't some sort of nameless temporary file be more indicated if a file has to be used?
Couldn't we use a constant string together with environment variables for `packages` (and possibly `vars`) instead?